src/components/posts/AddPostForm.js

```diff
@@ -13,20 +13,40 @@ function AddPostForm(props) {
     const enteredAvatar = avatarInputRef.current.value;
     const enteredContent = contentInputRef.current.value;
 
-    authorInputRef.current.value = "";
-    avatarInputRef.current.value = "#000000";
-    contentInputRef.current.value = "";
+    if(validation(enteredAuthor, enteredContent)){
+      authorInputRef.current.value = "";
+      avatarInputRef.current.value = "#000000";
+      contentInputRef.current.value = "";
+  
+      const postData = {
+        author: enteredAuthor,
+        avatar: enteredAvatar,
+        date: enteredDate,
+        content: enteredContent,
+        likes: 0,
+      };
+  
+      onAddPost(postData);
+    } else {
 
-    const postData = {
-      author: enteredAuthor,
-      avatar: enteredAvatar,
-      date: enteredDate,
-      content: enteredContent,
-      likes: 0,
-    };
+    }
+  }
 
-    props.onAddPost(postData);
+  function validation(Author, Content){
+    const inputRegex = /(^[A-z0-9À-ž.,'!?&$%()\s]+$)/;
+    if(!inputRegex.test(Author)){
```

Comment:
At the moment you accept all whitespace like enters for the Author, is that going to look strange if they use a whole bunch of whitespace?